src/App.tsx

```diff
@@ -15,49 +16,126 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortType {
+  NONE,
+  LENGTH,
+  ALPHABET,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  if (sortType === SortType.LENGTH) {
+    visibleGoods.sort((g1, g2) => {
+      return g1.length - g2.length;
+    });
+  }
+
+  if (sortType === SortType.ALPHABET) {
+    visibleGoods.sort((g1, g2) => {
+      return g1.localeCompare(g2);
+    });
+  }
+
+  if (isReversed) {
+    return visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+
+  // eslint-disable-next-line no-console
+  // console.log(sortType, isReversed);
+}
+
+export function App() {
+  const [sortType, sort] = useState(SortType.NONE);
+  const [isReversed, reverse] = useState(false);
```

Comment:
it's better to rename the function that updates the state and use _set_. it's standard name convention and is better for readability
```suggestion
  const [sortType, setSortType] = useState(SortType.NONE);
  const [isReversed, setIsReversed] = useState(false);
```